src/Orchard.Web/Modules/Orchard.Layouts/Services/ElementManager.cs

```diff
@@ -79,7 +79,7 @@ orderby elementDescriptor.DisplayText.Text
 
         public ElementDescriptor GetElementDescriptorByTypeName(DescribeElementsContext context, string typeName) {
             var elements = DescribeElements(context);
-            var element = elements.SingleOrDefault(x => x.TypeName == typeName);
+            var element = elements.SingleOrDefault(x => String.Equals(x.TypeName, typeName, StringComparison.OrdinalIgnoreCase));
```

Comment:
True, a dictionary would be much faster, provided that we cache it (otherwise there would be no gain, since it would to iterate over all elements to build the dictionary in the first place). We could use the type name (lower case) as the key.

I still don't buy the other argument against `SingleOrDefault`, because if you _know_ there will only ever be one match at the most, then it is a great help that it will throw an exception in case the set contains more than one result, as this would indicate a bug. Anything else would cover up the bug.